### PR TITLE
modify push notification settings + use fcm-django library

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

- swaps out `django-push-notifications` for [`fcm-django`](https://github.com/grafana/fcm-django). Again.. this is a fork of the parent repo for exactly the same reason.. the migrations point to `auth_user` without letting us use our own user model, this has been patched in the `grafana` fork. The reason why we are using `fcm-django` vs `django-push-notifications` is that the latter does not support the new FCM API, only the "legacy" API. The legacy FCM API does not support certain push notification settings that we would like to use.
- modifies the iOS/Android specific push notification settings
- adds a `flower` pod in the `docker-compose-developer.yml`, useful for debugging tasks locally
- sets the mobile app verification token TTL to 5 minutes when developing locally. The default of 1 minute makes working with device emulators really tricky..

## Other stuff

This PR also swaps out the base image in `engine/Dockerfile` from `python:3.9-alpine3.16` to `python:3.9-slim-buster`.

As to why.. in short, with the introduction of the `fcm-django` library there is now a peer-dependency on [`grpcio`](https://github.com/grpc/grpc) (which is used by `firebase_admin`.. which I am using in this PR to interact directly with Firebase Cloud Messaging (FCM)). `grpcio` does not publish wheels (read: compiled binaries) for the Alpine distro. It does publish wheels for Debian and hence `pip install -r requirements.txt` does not need to build this library from the source distribution.

This is a [known "issue"](https://github.com/grpc/grpc/issues/22815#issuecomment-1107874367) and the recommended solution in the community is to.. not use alpine.

These were the numbers, when building the image locally, in terms of image size and build time:

|                                             | Local image size (uncompressed  | Build time (may differ based on your network speed) |
| ------------------------- | -------------------------------------- | ---------- |
| `python:3.9-alpine3.16`   | 785MB  | 320s |
| `python:3.9-slim-buster` | 1.05GB  | 90s   |

### CODE REVIEW

**Comment by joeyorlando** [comment]
opened a new PR because I _think_ the caching for the old PR was causing the helm e2e tests to fail (despite them passing locally)

**Comment by iskhakov** [comment]
> swaps out django-push-notifications for [fcm-django](https://github.com/grafana/fcm-django). Again.. this is a fork of the parent repo for exactly the same reason.. the migrations point to auth_user without letting us use our own user model, this has been patched in the grafana fork. The reason why we are using fcm-django vs django-push-notifications is that the latter does not support the new FCM API, only the "legacy" API. The legacy FCM API does not support certain push notification settings that we would like to use.

I agree with that, fcm-django is the fork, so we can return to the original lbrary

File: engine/apps/mobile_app/fcm_relay.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+# from firebase_admin.messaging import Message
+# from fcm_django.models import FCMDevice
 from rest_framework import status
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
 REQUIRED_FIELDS = {"registration_ids", "notification", "data"}
 
 
+# TODO: update thie
```
Comment (iskhakov) [inline engine/apps/mobile_app/fcm_relay.py:10]: we need more explanation here

**Review by iskhakov — APPROVED** [review]

**Comment by imtoori** [comment]
A comment on my changes

I removed the `notification` payload from the android field because we're gonna use `data-only` messages so that we can execute some dart code to change the DND (Do Not Disturb) and volume settings before showing a critical notification.

For more information on data-only messages https://firebase.google.com/docs/cloud-messaging/concept-options

**Review by imtoori — APPROVED** [review]

**Pull request merged**